docusaurus/docs/dev-docs/plugins/documentation.md

```diff
@@ -51,332 +52,212 @@ This plugin comes with an interface that is available in your administration pan
-
-:::warning
-
-Do not change the `openapi` field of the `settings.json`
-
-:::
-
-:::note
-
-When you change a field in the settings.json file you need to manually restart your server.
-
-:::
-
 ### Create a new version of the documentation
 
-To create a new version of your documentation, you need to update the `version` key.
+To create a new version of your documentation, update the `version` key
 
-```
-{
-  "info": {
-    "version": "2.0.0"
-  }
-}
-```
+```js title="config/plugin.js"
 
-### Change the documentation path
+module.exports = {
+  documentation: {
+    enabled: true,
+    config: {
+      info: { version: "2.0.0" },
+    },
+  },
+};
+```
 
-To access your documentation on a custom path, you need to update the `path` key.
+### Indicate which plugins need documentation generated
 
-```
-{
-  "x-strapi-config": {
-    "path": "/documentation"
-  }
-}
-```
+If you want plugins to be included in documentation generation they should be included in the `plugins` array on the `x-strapi-config`. By default this array is initialized with `["upload", "users-permissions"]`.
 
-### Indicate which plugins documentation to generate
+Similarly, if you do not want plugins to be included in documentation generation, provide an empty array.
 
-To generate documentation for specific plugins, you need to indicate the list of all the plugins for which you wish to generate documentation. In order to do that you need to update the `pluginsForWhichToGenerateDoc` key. Leaving this key with an empty array `[]` means that no plugin documentation will be generated. If you wish to generate documentation for all plugins, you have to remove the key from the `settings.json` file.
+```js title="config/plugins.js"
 
-```
-{
-  "x-strapi-config": {
-    "pluginsForWhichToGenerateDoc": [
-      "email",
-      "upload",
-      "users-permissions"
-    ],
-  }
-}
+module.exports = {
+  documentation: {
+    enabled: true,
+    config: {
+      "x-strapi-config": {
+        // Default
+        plugins: ["upload", "users-permissions"],
+        // Custom
+        plugins: ["upload"],
+        // Do not generate for plugins
+        plugins: [],
+      },
+    },
+  },
+};
 ```
 
-The previous example generates documentation for the upload, email and users permissions (permissions and roles) plugins.
+## Overriding the generated documentation
 
-### Default Response
+### Excluding from generation
 
-Sometimes, an operation can return multiple errors with different HTTP status codes, but all of them have the same response structure. You can use the default response to describe these errors collectively, not individually. “Default” means this response is used for all HTTP codes that are not covered individually for this operation.
+If you simply want to exclude certain apis or plugins from being generated you can use the `excludeFromGeneration` function on the override service
```

Comment:
So for plugins should we use this or the plugins array?

Reply:
I guess in some cases you could do either or. If you wanted to provide an override from your plugin that mixes with the generated documentation then the plugin needs to be in the plugins array and excludeFromGeneration would not work for that case.

If you wanted to provide a plugin that completely replaces the generated documentation. You need to provide the override, exclude the plugin from generation, and provide the pluginOrigin so that the end user can opt-out of your plugin's documentation via the plugins array.

The real catch though is a plugin override that doesn't provide a pluginOrigin will cause the override to always run even if it is not in the plugins array. In that case though a user could go the extra mile and still get rid of the plugin using excludeFromGeneration.

It seems a bit confusing, you got any ideas? (for the docs or the code)

Reply:
Thanks that does make it clearer. I do think it would be worth mentioning this difference in the docs, maybe in some kind of widget

Reply:
as you want though, it doesn't block the PR

Reply:
Thanks for the question and discussion :)
If you feel like it's worth being highlighted, you can mention it:
- in a `:::note` callout if it does not "break" anything or doesn't cause a weird or unexpected behavior with the software,
- or in a `:::caution` callout if it's somewhat unexpected or tricky

I think it's not blocking the PR, either. So it's up to you. If it's not included in the PR we can still add it eventually, if users feedback mention this part was unclear :)